src/hooks/useApi.tsx

```diff
@@ -0,0 +1,71 @@
+import type { Api } from '@jellyfin/sdk';
+import type { UserDto } from '@jellyfin/sdk/lib/generated-client';
+import { ApiClient } from 'jellyfin-apiclient';
+import React, { createContext, FC, useContext, useEffect, useState } from 'react';
+
+import type ServerConnections from '../components/ServerConnections';
+import events from '../utils/events';
+import { toApi } from '../utils/jellyfin-apiclient/compat';
+
+interface ApiProviderProps {
+    connections: typeof ServerConnections
+}
+
+interface JellyfinApiContext {
+    __legacyApiClient__?: ApiClient
+    api?: Api
+    user?: UserDto
+}
+
+export const ApiContext = createContext<JellyfinApiContext>({});
+export const useApi = () => useContext(ApiContext);
+
+export const ApiProvider: FC<ApiProviderProps> = ({ connections, children }) => {
+    const [ legacyApiClient, setLegacyApiClient ] = useState<ApiClient>();
+    const [ api, setApi ] = useState<Api>();
+    const [ user, setUser ] = useState<UserDto>();
+
+    useEffect(() => {
+        connections.currentApiClient()
+            .getCurrentUser()
+            .then(newUser => udpateApiUser(null, newUser))
+            .catch(err => {
+                console.warn('[ApiProvider] Could not get current user', err);
+            });
```

Comment:
why?

Reply:
getCurrentUser() could be null if no stored server available

Reply:
Good catch. I didn't realize the apiclient would throw if there is no current user there.